spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -109,10 +113,19 @@ protected SparkSession lazySparkSession() {
     return lazySpark;
   }
 
-  protected Configuration lazyConf() {
+  protected Configuration lazyBaseConf() {
     if (lazyConf == null) {
       this.lazyConf = lazySparkSession().sparkContext().hadoopConfiguration();
     }
     return lazyConf;
   }
+
+  protected Configuration mergeIcebergHadoopConfs(Configuration baseConf, Map<String, String> options) {
+    Configuration resolvedConf = new Configuration(baseConf);
+    options.keySet().stream()
+        .filter(key -> key.startsWith("iceberg.hadoop"))
+        .filter(key -> baseConf.get(key) == null)
```

Comment:
Should this remove the option from the Configuration?

Reply:
I'm ok with any opinions about which options take precedence in different permutations. For example should the order of precedence be Spark Session -> Table Properties -> Data Source Options? Or in some other order?

Reply:
Quick ping here?

Reply:
I think order of precedence should be:
* Iceberg defaults
* Table configuration
* Write options

I'm not sure how to separate session from write options. I don't think we pull any configuration from the session to configure a specific write, do we?

Reply:
There isn't anything for the session, but there is some for the Spark Context. Perhaps one considers it as JVM-wide configuration? It's accessed by `SparkSession.sparkContext().hadoopConfiguration` - that's what `lazySparkSession()` returns right now.

Reply:
Also what are "Iceberg Defaults" in this case?

Reply:
Hadoop configuration overrides Iceberg defaults because Iceberg defaults are applied if nothing is set. Table properties will be applied to the configuration next to override, and then write options to override the table configuration. So the environment config is in between defaults and table config because the Configuration is how those things are passed.

Now that I look at this, I think that Iceberg table properties are currently highest precedence because the write options get applied to the configuration before the table properties do (so that they are set when the table is looked up). So we should either make the expectation that table properties will always win, or add the write properties a second time. Probably the latter option. What do you think?

Reply:
Iceberg defaults are the values that get filled in when nothing is set. They could be Parquet or some other default as well. I was thinking about Iceberg properties.

Reply:
I rewrote some of this in my latest push to reflect the suggested order of precedence given in [a few comments up](https://github.com/apache/incubator-iceberg/pull/7#discussion_r239177508). I think the precedence should have the Spark Context's and the write options take higher precedence than what's specified in table properties. Given that, merging hadoop configurations in the helper methods needs to have a flag knowing whether or not to override. Thoughts on the new approach?